symfony/plugins/orangehrmPimPlugin/Api/EmpUsTaxExemptionAPI.php

```diff
@@ -0,0 +1,241 @@
+ */
+
+namespace OrangeHRM\Pim\Api;
+
+use Exception;
+use OrangeHRM\Core\Api\CommonParams;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointResourceResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\ParameterBag;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\ResourceEndpoint;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Core\Traits\Service\ConfigServiceTrait;
+use OrangeHRM\Entity\EmpUsTaxExemption;
+use OrangeHRM\Pim\Api\Model\EmpUsTaxExemptionModel;
+use OrangeHRM\Pim\Service\EmpUsTaxExemptionService;
+
+class EmpUsTaxExemptionAPI extends Endpoint implements ResourceEndpoint
+{
+    use ConfigServiceTrait;
+
+    public const PARAMETER_FEDERAL_STATUS = 'federalStatus';
+    public const PARAMETER_FEDERAL_EXEMPTIONS = 'federalExemptions';
+    public const PARAMETER_TAX_STATE_CODE = 'taxStateCode';
+    public const PARAMETER_STATE_STATUS = 'stateStatus';
+    public const PARAMETER_STATE_EXEMPTIONS = 'stateExemptions';
+    public const PARAMETER_UNEMPLOYMENT_STATE_CODE = 'unemploymentStateCode';
+    public const PARAMETER_WORK_STATE_CODE = 'workStateCode';
+
+    /**
+     * @var EmpUsTaxExemptionService|null
+     */
+    protected ?EmpUsTaxExemptionService $empUsTaxExemptionService = null;
+
+    /**
+     * @return EmpUsTaxExemptionService
+     */
+    public function getEmpUsTaxExemptionService(): EmpUsTaxExemptionService
+    {
+        if (!$this->empUsTaxExemptionService instanceof EmpUsTaxExemptionService) {
+            $this->empUsTaxExemptionService = new EmpUsTaxExemptionService();
+        }
+        return $this->empUsTaxExemptionService;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getOne(): EndpointResourceResult
+    {
+        $empNumber = $this->getRequestParams()->getInt(
+            RequestParams::PARAM_TYPE_ATTRIBUTE,
+            CommonParams::PARAMETER_EMP_NUMBER
+        );
+        $empUsTaxExemption = $this->getEmpUsTaxExemptionService()->getEmpUsTaxExemptionDao()->getEmployeeTaxExemption($empNumber);
+        $this->throwRecordNotFoundExceptionIfNotExist($empUsTaxExemption, EmpUsTaxExemption::class);
+
+        return new EndpointResourceResult(
+            EmpUsTaxExemptionModel::class, $empUsTaxExemption,
+            new ParameterBag([CommonParams::PARAMETER_EMP_NUMBER => $empNumber])
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetOne(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(
+                CommonParams::PARAMETER_EMP_NUMBER,
+                new Rule(Rules::IN_ACCESSIBLE_EMP_NUMBERS)
+            ),
+            new ParamRule(
+                CommonParams::PARAMETER_ID,
+                new Rule(Rules::BETWEEN, [0, 100])
```

Comment:
rule is wrong. Since id hardcoded in route

Reply:
added equal[0]